exercises/rna-transcription/example/RNA.ceylon

```diff
@@ -0,0 +1,13 @@
+class Error() {}
+
+String|Error transcription(String dna) {
+  return dna.fold("" of String|Error)((String|Error r,
+                                       Character    c) =>   if(is Error r)
+                                                          then(r)
+                                                          else(switch(c)
```

Comment:
I would have thought it would be valid (why not the `switch` but you can remove the parentheses on the `else`?) but it isn't.